source/fundamentals/data-formats/document-data-format-extended-json.txt

```diff
@@ -0,0 +1,250 @@
+                     "\"myNumber\": {\"$numberLong\": \"4794261\" }}}";
+
+   Document doc = new Document(Document.parse(ejsonStr));
+   System.out.println(doc);
+
+The output of the code above resembles the following:
+
+.. code-block:: none
+   :copyable: False
+
+   Document{{_id=507f1f77bcf86cd799439011, myNumber=4794261}}
+
+For more information on the driver document classes, see our Fundamentals page
+on :doc:`Documents </fundamentals/data-formats/documents>`.
+
+For additional information on the classes and methods mentioned on this
+section, see the following API documentation:
+
+Write Extended JSON
+-------------------
+
+You can convert Java objects into Extended JSON with the BSON library by using
+the :java-docs:`JsonWriter </apidocs/bson/org/bson/json/JsonWriter.html>`
+class. To construct an instance of ``JsonWriter``, pass a
+subclass of a Java ``Writer`` to specify how you want to output the Extended
+JSON. You can optionally pass a :java-docs:`JsonWriterSettings </apidocs/bson/org/bson/json/JsonWriterSettings.html>`
+instance to specify options such as the Extended JSON format. By default, the
+``JsonWriter`` uses the "Strict" Extended JSON format to guarantee backwards
+compatibility.
+
+The following code example shows how you can use ``JsonWriter`` to create an
+Extended JSON string and output it to ``System.out``. We specify the format
+by passing the :java-docs:`outputMode() </apidocs/bson/org/bson/json/JsonWriterSettings.Builder.html#outputMode(org.bson.json.JsonMode)>`
+builder method the ``JsonMode.EXTENDED`` constant:
+
+.. code-block:: java
+
+   JsonWriterSettings settings = JsonWriterSettings.builder().outputMode(JsonMode.EXTENDED).build();
+
+   try (JsonWriter jsonWriter = new JsonWriter(new BufferedWriter(new OutputStreamWriter(System.out)), settings)) {
+       jsonWriter.writeStartDocument();
+       jsonWriter.writeObjectId("_id", new ObjectId("507f1f77bcf86cd799439012"));
+       jsonWriter.writeInt64("myNumber", 11223344);
+       jsonWriter.writeEndDocument();
+       jsonWriter.flush();
+   }
+
+The output of this code example resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   {"_id": {"$oid": "507f1f77bcf86cd799439012"}, "myNumber": {"$numberLong": "11223344"}}
+
+To write Extended JSON from an instance of a ``Document``, you can call the
+``toJson()`` method, passing it an instance of ``JsonWriterSettings`` to 
+specify the Extended JSON format. In this example, we use the "Shell" format.
+
+.. code-block:: java
+
+   Document myDoc = new Document();
+   myDoc.append("_id", new ObjectId("507f1f77bcf86cd799439012")).append("myNumber", 11223344);
+
+   JsonWriterSettings settings = JsonWriterSettings.builder().outputMode(JsonMode.SHELL).build();
+   System.out.println(doc.toJson(settings));
+
+The output of this code example resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   {"_id": ObjectId("507f1f77bcf86cd799439011"), "myNumber": NumberLong(4794261)}
```

Comment:
It might be good to explain how to add custom conversion of bson types.  

One such example can be found here: https://github.com/mongodb/mongo-kafka/blob/r1.3.0/src/main/java/com/mongodb/kafka/connect/source/json/formatter/SimplifiedJson.java

That takes the Relaxed output mode and also changes ObjectId's to output just hex strings, dates to be outputted as ISO date strings etc..

Reply:
Sounds good, I'll add a section. Is this a frequently-asked question from the community/customers?

Reply:
Addressed in f154afda39c6d5d83d850a77618fb0a8673b4eb5